pinot-common/src/test/java/org/apache/pinot/common/metrics/PinotJMXToPromMetricsTest.java

```diff
@@ -0,0 +1,379 @@
+    List<PromMetric> promMetrics;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+      for (String meterType : TIMER_TYPES) {
+        Assert.assertTrue(
+            promMetrics.contains(PromMetric.withName(exportedMetricPrefix + exportedTimerPrefix + "_" + meterType)));
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void assertTimerExportedCorrectly(String exportedTimerPrefix, List<String> labels,
+      String exportedMetricPrefix) {
+    List<PromMetric> promMetrics;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+      for (String meterType : METER_TYPES) {
+        Assert.assertTrue(promMetrics.contains(
+                PromMetric.withNameAndLabels(exportedMetricPrefix + exportedTimerPrefix + "_" + meterType, labels)),
+            exportedTimerPrefix);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void assertMeterExportedCorrectly(String exportedMeterPrefix, String exportedMetricPrefix) {
+    List<PromMetric> promMetrics;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    for (String meterType : METER_TYPES) {
+      Assert.assertTrue(
+          promMetrics.contains(PromMetric.withName(exportedMetricPrefix + exportedMeterPrefix + "_" + meterType)),
+          exportedMeterPrefix);
+    }
+  }
+
+  protected void assertMeterExportedCorrectly(String exportedMeterPrefix, List<String> labels,
+      String exportedMetricPrefix) {
+    List<PromMetric> promMetrics;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+      for (String meterType : METER_TYPES) {
+        Assert.assertTrue(promMetrics.contains(
+                PromMetric.withNameAndLabels(exportedMetricPrefix + exportedMeterPrefix + "_" + meterType, labels)),
+            exportedMeterPrefix);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected List<PromMetric> parseExportedPromMetrics(String response)
+      throws IOException {
+
+    List<PromMetric> exportedPromMetrics = new ArrayList<>();
+
+    try (BufferedReader reader = new BufferedReader(new StringReader(response))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        if (line.startsWith("pinot_")) {
+          exportedPromMetrics.add(PromMetric.fromExportedMetric(line));
+        }
+      }
+      reader.close();
+      return exportedPromMetrics;
+    }
+  }
+
+  protected abstract SimpleHttpResponse getExportedPromMetrics();
+
+  protected abstract void timerTest();
+
+  protected abstract void gaugeTest();
+
+  protected abstract void meterTest();
+
+  public static class PromMetric {
```

Comment:
nit: let's try to declare classes after all methods, not in between

Reply:
Honestly, I'd rather have these added to the auto-formatter/checkstyle to detect this. We currently don't have any check that checks the order of fields/methods, etc. I'll fix this though.

Reply:
This is done